src/tlo/methods/contraception.py

```diff
@@ -644,27 +646,103 @@ def select_contraceptive_following_birth(self, mother_id, mother_age):
 
-    def get_item_code_for_each_contraceptive(self):
-        """Get the item_code for each contraceptive and for contraceptive initiation."""
-        # TODO: update with optional items (currently all considered essential)
-
-        get_items_from_pkg = self.sim.modules['HealthSystem'].get_item_codes_from_package_name
+    def get_item_codes_for_each_contraceptive_and_probs_for_cons_alternatives(self):
+        """Get the item_code and numbers of units per case for each contraceptive and for contraceptive initiation.
+        Save the probs for consumable alternatives."""
+
+        # ### Get item codes and number of units per case from package name
+        # (note: this version does not deal with consumable alternatives)
+        # get_items_from_pkg = self.sim.modules['HealthSystem'].get_item_codes_from_package_name
+
+        # _cons_codes = dict()
+        # # items for each method that requires an HSI to switch to
+        # _cons_codes['pill'] = get_items_from_pkg('Pill')
+        # _cons_codes['male_condom'] = get_items_from_pkg('Male condom')
+        # _cons_codes['other_modern'] = get_items_from_pkg('Female Condom')
+        # # NB. The consumable female condom is used for the contraceptive state of "other_modern method"
+        # _cons_codes['IUD'] = get_items_from_pkg('IUD')
+        # _cons_codes['injections'] = get_items_from_pkg('Injectable')
+        # _cons_codes['implant'] = get_items_from_pkg('Implant')
+        # _cons_codes['female_sterilization'] = get_items_from_pkg('Female sterilization')
+        # assert set(_cons_codes.keys()) == set(self.states_that_may_require_HSI_to_switch_to)
+        # # items used when initiating a modern reliable method after not using or switching from non-reliable method
+        # _cons_codes['co_initiation'] = get_items_from_pkg('Contraception initiation')
+
+        # ### Get item codes from item names and define number of units per case here
+        get_item_code = self.sim.modules['HealthSystem'].get_item_code_from_item_name
 
         _cons_codes = dict()
-        # items for each method that requires an HSI to switch to
-        _cons_codes['pill'] = get_items_from_pkg('Pill')
-        _cons_codes['male_condom'] = get_items_from_pkg('Male condom')
-        _cons_codes['other_modern'] = get_items_from_pkg('Female Condom')
-        # NB. The consumable female condom is used for the contraceptive state of "other_modern method"
-        _cons_codes['IUD'] = get_items_from_pkg('IUD')
-        _cons_codes['injections'] = get_items_from_pkg('Injectable')
-        _cons_codes['implant'] = get_items_from_pkg('Implant')
-        _cons_codes['female_sterilization'] = get_items_from_pkg('Female sterilization')
+        _cons_alternatives_probs = dict()
+        # # items for each method that requires an HSI to switch to
+        _cons_codes['pill'] =\
+            {get_item_code("Levonorgestrel 0.0375 mg, cycle"): 21*3.75,
```

Comment:
int?

Reply:
15 packets administrated per year according to OHT 2016, hence average per administration (every 3 months) is 3.75 packets.